torch/distributed/fsdp/flat_param.py

```diff
@@ -75,6 +75,13 @@
 # special cases such as for high CPU overhead or for intentionally bypassing
 # checks in the overrides, we may use 'unsafe'.
 _FSDP_USE_UNSAFE_SETATTR = "FSDP_USE_UNSAFE_SETATTR"
+# Environment variable toggling whether to check for parameter/gradient
+# writeback in case their storages change after FSDP initialization
+# We should check by default since it prevents silent correctness errors, but
+# since such changes are atypical, we may want to skip the check to save CPU
+# overhead, especially since the check happens in the pre-forward and
+# pre-backward each iteration.
+_FSDP_SKIP_WRITEBACK_CHECK = "FSDP_SKIP_WRITEBACK_CHECK"
```

Comment:
can you provide an example of the silent correctness issue that could surface so we can assess the risk for different workloads?

Reply:
The user pretty much has to explicitly set `param.data = ...` or replace the parameter variable like `module.param = new_param`. I am not aware of any other ways, but I added the check in the original `use_orig_params=True` implementation since I was not sure if I was missing anything.